src/zenml/cli/stack_components.py

```diff
@@ -1151,100 +1198,135 @@ def deploy_stack_component_command(
         Args:
             name: Name of the component to register.
             flavor: Flavor of the component to register.
-            cloud: Cloud provider to use to deploy the stack component.
-            args: Additional arguments to pass to the component.
+            provider: Cloud provider (or local) to use to deploy the stack
+            component.
+            region: Region to deploy the stack component to.
+            debug_mode: Whether to deploy the stack component in debug mode.
+            tags: Tags to be added to the component.
+            extra_config: Extra configuration values to be added to the
 
         Raises:
             error: If the component type is not supported.
         """
-        # generate a python dict with the above structure
-        # and then use it to check if the flavor is valid
-        # for the given component type
-        allowed_flavors = {
-            "experiment_tracker": ["mlflow"],
-            "model_deployer": ["seldon", "kserve"],
-            "artifact_store": ["s3", "gcp", "minio"],
-            "container_registry": ["gcp", "aws"],
-            "orchestrator": [
-                "kubernetes",
-                "kubeflow",
-                "tekton",
-                "sagemaker",
-                "vertex",
-            ],
-            "step_operator": ["sagemaker", "vertex"],
-        }
+        client = Client()
+        try:
+            # raise error if user already has a component with the same name
+            client.get_stack_component(
```

Comment:
This is fine, but I'd go with `list_stack_components` + `if`

Reply:
For my understanding, why do you prefer the larger call? Are you concerned that maybe it won't get the stack component?